### PR TITLE
chore: Split share modal component into smaller components

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

Split share modal component into smaller components

#### PR fixes following issue(s)
Fixes [#27851](https://github.com/appsmithorg/appsmith/issues/27851) [#27671](https://github.com/appsmithorg/appsmith/issues/27671)

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [x] Cypress

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/ce/pages/workspace/helpers.ts [hunk 1]
```diff
@@ -12,6 +12,19 @@ export type InviteUsersToWorkspaceFormValues = {
   usersByRole: InviteUsersToWorkspaceByRoleValues[];
 };
 
+export type InviteUsersProps = {
+  roles?: any;
```
Comment (KelvinOm) [inline app/client/src/ce/pages/workspace/helpers.ts:16]: Can we type this all correctly? Let's get rid of `any`.

Comment (ankitakinger) [reply app/client/src/ce/pages/workspace/helpers.ts:16]: this is still a WIP PR. Let me change it to draft.

File: app/client/src/ce/pages/workspace/helpers.ts [hunk 2]
```diff
@@ -12,6 +13,19 @@ export type InviteUsersToWorkspaceFormValues = {
   usersByRole: InviteUsersToWorkspaceByRoleValues[];
 };
 
+export type InviteUsersProps = {
```
Comment (dipyamanbiswas07) [inline app/client/src/ce/pages/workspace/helpers.ts:16]: why is the type in helpers? should it be present in /types folder?

**Review by dipyamanbiswas07 — APPROVED** [review]
LGTM

File: app/client/src/ce/pages/workspace/helpers.ts (hunk 2 above)
Comment (ankitakinger) [reply app/client/src/ce/pages/workspace/helpers.ts:16]: Because the other types were already here, so I just kept it at the same place

**Review by dipyamanbiswas07 — APPROVED** [review]

**Pull request merged**